src/ui/component/modal/modal.tsx

```diff
@@ -0,0 +1,76 @@
+import { type ReactNode, type FC, useCallback } from 'react'
+import { createPortal } from 'react-dom'
+import classNames from 'classnames'
+import { CSSTransition } from 'react-transition-group'
+import { useOnKeyboard } from '@/ui/hook/useOnKeyboard'
+import './modal.scss'
+
+type ModalProps = {
+  isOpen: boolean
+  onClose: () => void
+  closeOnEsc?: boolean
+  motionPreset?: 'zoom' | 'none'
+  isCentered?: boolean
+  children?: ReactNode
+  classes?: {
+    main?: string
+    overlay?: string
+  }
+}
+
+// eslint-disable-next-line max-lines-per-function
+export const Modal: FC<ModalProps> = ({
+  isOpen,
+  onClose,
+  closeOnEsc = true,
+  motionPreset = 'zoom',
+  isCentered = true,
+  children,
+  classes
+}) => {
+  const handleEscape = useCallback(
+    (event: KeyboardEvent) => {
+      if (closeOnEsc && event.key === 'Escape') {
+        onClose()
+      }
+    },
+    [onClose, closeOnEsc]
+  )
+
+  useOnKeyboard(handleEscape)
+
+  const handleOverlayClick = useCallback(() => {
+    onClose()
+  }, [onClose])
+
+  const containerElement = document.getElementById('modal')
```

Comment:
Instead of creating a container dedicated to the modal, we would prefer to select the highest html element in the DOM hierarchy that can received themed style. In other word, the immediate child element of the div with the `theme--${theme}` class.
In our case, that would be the div with the following class:`.okp4-dataverse-portal-main-layout`.

You can reference it at the beginning of your component s follow:
```tsx
   const containerRef = useRef<HTMLElement | null>(null)

    useEffect(() => {
        containerRef.current = document.querySelector('.okp4-dataverse-portal-main-layout')
      }, [])


    // other code


    // don't forget to add it as `nodeRef` to avoid `findDOMNode` errors
    <CSSTransition
      nodeRef={containerRef}
      // other props
     
```

Reply:
Good idea, I fixed it now :)